client/asset/eth/eth.go

```diff
@@ -871,6 +894,12 @@ func (w *ETHWallet) Reconfigure(ctx context.Context, cfg *asset.WalletConfig, cu
 		gasFeeLimit = defaultGasFeeLimit
 	}
 
+	// For now, we only are supporting multiRPCClient nodes. If we re-implement
+	// P2P nodes, we'll have to add protection to the node field to allow for
+	// reconfiguration of type.
+	// We also need to consider how to handle the pendingTxs if we switch node
+	// types, since right now we only use that map for multiRPCClient.
```

Comment:
You could only change the multirpcclient to return the pending txn and I don't think this would be a problem. Will there be other uses for the new `mempoolFetcher` interface?

Reply:
I think it makes sense to only provide methods which the node supports. We could handling pending tx caching on the node side, as you've suggested, but it means the node would need to understand the format of the txs for swaps, redeems, refunds, transfers, which breaks a lot of assumptions and our scheme of separating concerns.

Reply:
I had the same thought as @JoeGruffins, so I'd like to comment on this in case it clarifies.

The concerns you are referring to are if the to/from accounts etc are the user, if it comes from the balance, or was in/goes to the swap contract, etc.  The names of the `mempoolFetcher` interface (and the `pendingTransactions() ([]*types.Transaction, error)` method) are broad because they intend to fetch whatever transactions, no assumptions about ownership etc.  That's not something a mock method that `multiRPCClient` can satisfy.